### ISSUE TITLE
LV_DRAW_BUF_INIT_STATIC broken for LV_COLOR_FORMAT_L8

### ISSUE THREAD

**Issue by capricorn-one** [kisvegabor/lvgl#6718]
### LVGL version

v9.1.1.dev

### What happened?

Error: argument of type "uint32_t" (aka "unsigned int") is incompatible with parameter of type "lv_color_format_t"C/C++(167)

Not sure if this works for any other color formats, don't see how it could tbh, but easy fix for now is casting header->cf as a lv_color_format_t in the define.

`#define LV_DRAW_BUF_INIT_STATIC(name) \
    do { \
        lv_image_header_t * header = &name.header; \
        lv_draw_buf_init(&name, header->w, header->h, (lv_color_format_t)header->cf, header->stride, buf_##name, sizeof(buf_##name)); \
        lv_draw_buf_set_flag(&name, LV_IMAGE_FLAGS_MODIFIABLE); \
    } while(0)
`

### How to reproduce?

```
LV_DRAW_BUF_DEFINE_STATIC(buffer, RPM_CANVAS_WIDTH, RPM_CANVAS_HEIGHT, LV_COLOR_FORMAT_L8);
LV_DRAW_BUF_INIT_STATIC(buffer);
```

**Comment by kisvegabor** [comment]
I see no error or warning for it with GCC but opened a PR to fix it here: #6729

**Comment by capricorn-one** [comment]
yeah... I'm not surprised to hear that. I'm using platform.io Arduino framework, probably has more to do with that. But given other people will use it, and doesn't compile as is, probably worth fixing. Thanks for putting in the PR!
